compilation error with OCCT >6.9.1

In the Wiki you reommend VisuaStudio 2010. However you use newer capabilities which where introduced with VS2015.

Namely:
UniquePtr.h(25) uses [[deprecated]] - this does not compilie with VS2013
Cache.h(39) uses defaulted copy constructor - this does not compile with VS2013.

.....\tigl\src\Cache.h(39): error C2610: 'tigl::ScopeLockedReference<T>::ScopeLockedReference(tigl::ScopeLockedReference<T> &&)' : is not a special member function which can be defaulted (C:\data\gitRepos\tigl\src\CCPACSXsiIsoLine.cpp)
1>          c:\data\gitrepos\tigl\src\Cache.h(47) : see reference to class template instantiation 'tigl::ScopeLockedReference<T>' being compiled

The deprecated flag can be removed to compile.
However, I am a bit at a loss how to convert the "default" to what should happen in the constructor.

In Cache.h, replaced ScopeLockedReference(ScopeLockedReference&&) = default; with

`		ScopeLockedReference(ScopeLockedReference&& ref) :
			m_reference(std::move(ref.m_reference)),
			m_mutex(std::move(ref.m_mutex))
		{}`

No clue if this is correct. Now other errors appear:

`......\tigl\src\.\generated/TixiHelper.h(53): error C2144: syntax error : 'auto' should be preceded by ';' (......\tigl\src\configuration\CCPACSConfiguration.cpp)
......\tigl\src\.\generated/TixiHelper.h(53): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int ((......\tigl\src\configuration\CCPACSConfiguration.cpp)`

I could get a hold of Visual Studio 2019 Community. Here cmake does not generate the compiler flags. But I could reuse the ones from VS2013.

Also, for the java bindings I use openjdk 11. For that to compile I had to copy some files from cmake install dir to the make dir in the project, namely:
`FindJava.cmake
CMakeFindJavaCommon.cmake
FindPackageHandleStandardArgs.cmake
FindPackagemessages.cmake`
Maybe update your cmake?

Back to the compile problem. I reverted my change. Now with VS2019 the error is:
`2>......\tigl\src\math\CTiglProjectPointOnCurveAtAngle.cpp(228,40): error C2664:  "math_BFGS::math_BFGS(const math_BFGS &)" : Konvertierung von Argument 1 von "`anonymous-namespace'::ProjectObjective" in "const Standard_Integer" nicht möglich`

Also I am not to conviced line 193 in CTiglProjectPointOnCurveAtAngle.cpp is correct:
`void CTiglProjectPointOnCurveAtAngle::CTiglProjectPointOnCurveAtAngle::Compute() const`

--- a/src/math/CTiglProjectPointOnCurveAtAngle.cpp
+++ b/src/math/CTiglProjectPointOnCurveAtAngle.cpp
@@ -30,6 +30,8 @@
 
 #include <algorithm>
 
+#include <Standard_Version.hxx>
+
 
 namespace 
 {
@@ -190,7 +192,7 @@ int tigl::CTiglProjectPointOnCurveAtAngle::NbPoints() const
     return static_cast<int>(resultParameter.size());
 }
 
-void CTiglProjectPointOnCurveAtAngle::CTiglProjectPointOnCurveAtAngle::Compute() const
+void tigl::CTiglProjectPointOnCurveAtAngle::Compute() const
 {
     if (m_hasComputed) {
         return;
@@ -225,7 +227,12 @@ void tigl::CTiglProjectPointOnCurveAtAngle::FindPoint(double ustart) const
     xstart(1) =  ustart;
 
     // TODO: how to use bounded optimization?
+#if OCC_VERSION_HEX >= VERSION_HEX_CODE(6,9,1)
+    math_BFGS optimizer(objFun.NbVariables());
+    optimizer.Perform(objFun,xstart);
+#else
     math_BFGS optimizer(objFun, xstart);
+#endif
     
     if (!optimizer.IsDone()) {
         return;

